File: src/main/java/com/github/rabend/generators/ObjectGenerator.java
package com.github.rabend.generators;

import com.fasterxml.jackson.databind.JsonNode;

import java.util.HashMap;
import java.util.Map;
import java.util.stream.Collectors;

public class ObjectGenerator extends AbstractValueGenerator {
    @Override
    public String generateRandomValue(final JsonNode node) {
        Map<String, AbstractValueGenerator> propertyToGenerator = new HashMap<>();
        if (node.has("properties")) {

            JsonNode propertiesNode = node.get("properties");

            propertiesNode.fields()
                    .forEachRemaining(propAndNode ->
                            propertyToGenerator.put(propAndNode.getKey(),
                                    ValueGeneratorsLookup.getGeneratorForType(propAndNode.getValue().get("type").asText())));

            return propertyToGenerator.entrySet()
                    .stream()
                    .map(propToGen -> {
                        String propName = propToGen.getKey();
                        return createKey(propName) + propToGen.getValue().generateRandomValue(propertiesNode.get(propName));
                    })
                    .collect(Collectors.joining(",", "{", "}"));
        } else {
            return "{}";
        }
    }

    private String createKey(final String prop) {
        return "\"" + prop + "\":";
    }
}

File: src/main/java/com/github/rabend/generators/ObjectGenerator.kt
package com.github.rabend.generators

import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.jsonPrimitive

class ObjectGenerator: AbstractValueGenerator {
    override fun generateRandomValue(baseObject: JsonObject): JsonElement {
        return if (baseObject.containsKey("properties")) {
            val propertiesNode = baseObject["properties"] as JsonObject
            val propertyToGenerator = propertiesNode.toList().associate {
                it.first to ValueGeneratorsLookup.getGeneratorForType(
                    (it.second as JsonObject)["type"]!!.jsonPrimitive.content
                )
            }
            val randomValuesMap =
                propertyToGenerator.entries.associate { (propName, value): Map.Entry<String, AbstractValueGenerator> ->
                    propName to value.generateRandomValue(propertiesNode[propName] as JsonObject)
                }
            JsonObject(randomValuesMap)
        } else {
            JsonObject(emptyMap())
        }
    }
}
